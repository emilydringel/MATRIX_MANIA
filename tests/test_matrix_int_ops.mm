def int main(){
    matrix<int> m1 = [1,2,3;4,5,6];
    matrix<int> m2 = [6,5,4;3,2,1];
    printm(m1+m2);
    printm(m1-m2);
    printm(m1*3);
    printm(3*m1);
    printm(m1*3.5);
    printm(3.5*m1);
    matrix<int> m3 = [1,2;3,4;5,6];
    printm(m1*m3);
    print(m1==m3);
    int x = (m1!=m3);
    print(x);
}