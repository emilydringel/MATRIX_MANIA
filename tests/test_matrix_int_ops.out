7 7 7
7 7 7
-5 -3 -1
1 3 5
3 6 9
12 15 18
3 6 9
12 15 18
3 6 9
12 15 18
3 6 9
12 15 18
22 28
49 64
0
1